src/meta/src/barrier/recovery.rs

```diff
@@ -403,9 +403,11 @@ impl GlobalBarrierManager {
                     };
 
                     // Reset all compute nodes, stop and drop existing actors.
-                    self.reset_compute_nodes(&info).await.inspect_err(|err| {
-                        warn!(error = %err.as_report(), "reset compute nodes failed");
-                    })?;
+                    self.reset_compute_nodes(&info, prev_epoch.value().0)
```

Comment:
In order for this PR to work, we need to ensure that `prev_epoch == latest MCE`. Currently this is guaranteed because the `prev_epoch` is get from `self.context.hummock_manager.latest_snapshot().committed_epoch` before calling `fn recovery`.

How about making it more explicitly by getting `self.hummock_manager.latest_snapshot().committed_epoch` inside the implementation of `fn recovery`?